sphinx/source/docs/dev_guide/bokehjs.rst

```diff
@@ -3,220 +3,312 @@
+more details, see the rules defined in :bokeh-tree:`bokehjs/eslint.json`.
 
-Requirements
-~~~~~~~~~~~~
+.. tip::
+  If you use VSCode, you can use the following configuration for your workspace
+  to use ESLint directly in the editor:
 
-* node 14+
-* npm 7.4+ (most recent version)
-* chrome/chromium browser 94+ or equivalent
+  .. code-block:: json
 
-You can install nodejs with conda:
+        "eslint.format.enable": true,
+        "eslint.lintTask.enable": true,
+        "eslint.debug": false,
+        "eslint.quiet": false,
+        "eslint.options": {
+          "cache": true,
+          "extensions": [".ts"],
+          "overrideConfigFile": "./eslint.json"
+        },
+        "eslint.workingDirectories": [
+          "./bokehjs"
+        ]
 
-.. code-block:: sh
+  This requires the `ESLint extension for VSCode`_ and ESLint version 8 or above
+  to be installed.
 
-    $ conda install -c conda-forge nodejs
+.. _contributor_guide_bokehjs_requirements:
 
-or follow the official installation `instructions <https://nodejs.org/en/download/>`_.
+Development requirements
+------------------------
 
-Upgrade your npm after installing or updating nodejs, or whenever asked by npm:
+To build and test BokehJS locally, follow the instructions in
+:ref:`contributor_guide_setup`. This way, all required packages
+should be installed and configured on your system.
 
-.. code-block:: sh
+Developing BokehJS requires the following minimum versions:
 
-    $ npm install -g npm@7
+* Node.js 14+
+* npm 7.4+
+* Chrome/Chromium browser 94+ or equivalent
 
-Officially supported platforms are as follows:
+Bokeh officially supports the following platforms for development and testing:
 
 * Linux Ubuntu 20.04+ or equivalent
 * Windows 10 (or Server 2019)
 * MacOS 10.15
 
-BokehJS can be developed on different platforms and versions of aforementioned
-software, but results may vary, especially when it comes to testing (visual
-testing in particular).
+It is possible to work on BokehJS using different platforms and versions.
+However, things might not work as intended, and some tests will not work.
 
-Building
-~~~~~~~~
+.. _contributor_guide_bokehjs_building:
 
-BokehJS's build is maintained by using an in-house tool that visually resembles
-gulp. All commands start with ``node make`` (don't confuse this with GNU make).
+Building BokehJS
+----------------
 
-Most common commands:
+For building, BokehJS relies on a custom tool similar to gulp_. All
+commands start with ``node make``.
 
-* ``node make build``
-* ``node make test``
-* ``node make lint``
+Use ``node make help`` to list all available commands for the BokehJS build
+system. These are the most common commands:
 
-Use ``node make help`` to list all available commands.
+* ``node make build``: Builds the entire library, including the extensions
+  compiler.
+* ``node make dev``: Builds the library without the extensions compiler. This is
```

Comment:
@mattpap Thanks for your comment on the difference between `node make build` and `node make dev`, I updated the two sentences accordingly. However, I'm afraid I don't quite know what the extensions compiler is. Could you explain that in a sentence or two, or point me to the corresponding lines in the code?